proto/destination.proto

```diff
@@ -107,3 +108,13 @@ message TlsIdentity {
 message NoEndpoints {
   bool exists = 1;
 }
+
+// A hint of what protocol the service knows. The default value is
+// for the `hint` field to be not be set, essentially meaning "unknown".
+message ProtocolHint {
+  oneof protocol {
+    H2 h2 = 1;
```

Comment:
Could you please document what this means? AFAICT it means "supports our proprietary H2 upgrade mechanism" which implies understanding of some proprietary HTTP headers.

Reply:
I've added a general comment about the `h2` field, without going into too much detail, as @olix0r seemed to want to allow some vagueness.